src/sonic-bgpcfgd/tests/test_allow_list.py

```diff
@@ -0,0 +1,296 @@
+from app.allow_list import BGPAllowListMgr
+from app.directory import Directory
+from app.template import TemplateFabric
+
+from mock import MagicMock, patch
+import pytest
+
+
+def new_run_command(cmd):
+    return 0, "", ""
+
+global_constants = {
+    "bgp": {
+        "allow_list": {
+            "enabled": True,
+            "default_pl_rules": {
+                "v4": [ "deny 0.0.0.0/0 le 17" ],
+                "v6": [
+                    "deny 0::/0 le 59",
+                    "deny 0::/0 ge 65"
+                ]
+            }
+        }
+    }
+}
+
+#@pytest.mark.skip()
+@patch('app.allow_list.run_command', side_effect=new_run_command)
+def test_set_handler_with_community(mock_run_command):
+    test_set_handler_with_community.push_list_called = False
+    def push_list(args):
+        test_set_handler_with_community.push_list_called = True
+        assert args == [
+            'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 10 deny 0.0.0.0/0 le 17',
+            'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 20 permit 10.20.30.0/24 ge 25',
+            'ip prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4 seq 30 permit 30.50.0.0/16 ge 17',
+            'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 10 deny 0::/0 le 59',
+            'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 20 deny 0::/0 ge 65',
+            'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 30 permit fc00:20::/64 ge 65',
+            'ipv6 prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6 seq 40 permit fc00:30::/64 ge 65',
+            'bgp community-list standard COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020 permit 1010:2020',
+            'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V4 permit 10',
+            ' match ip address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V4',
+            ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+            'route-map ALLOW_LIST_DEPLOYMENT_ID_5_V6 permit 10',
+            ' match ipv6 address prefix-list PL_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020_V6',
+            ' match community COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_5_COMMUNITY_1010:2020',
+        ]
+    cfg_mgr = MagicMock()
+    cfg_mgr.update.return_value = None
+    cfg_mgr.push_list = push_list
+    cfg_mgr.get_text.return_value = []
+    common_objs = {
+        'directory': Directory(),
+        'cfg_mgr':   cfg_mgr,
+        'tf':        TemplateFabric(),
+        'constants': global_constants,
+    }
+    mgr = BGPAllowListMgr(common_objs, "CONFIG_DB", "BGP_ALLOWED_PREFIXES")
+    mgr.set_handler("DEPLOYMENT_ID|5|1010:2020", {
+        "prefixes_v4": "10.20.30.0/24,30.50.0.0/16",
+        "prefixes_v6": "fc00:20::/64,fc00:30::/64",
```

Comment:
do you check if there is ipv4 prefix in prefixes_v6 and vice versa?

Reply:
Yes. I check it here:
[v4](https://github.com/Azure/sonic-buildimage/pull/5309/files#diff-5857637ca540593180dbfe7a81cdd86fR82)
[v6](https://github.com/Azure/sonic-buildimage/pull/5309/files#diff-5857637ca540593180dbfe7a81cdd86fR88)

Reply:
i think test case to check that.

Reply:
Done

Reply:
which test case contains this? i cannot find it.

Reply:
https://github.com/Azure/sonic-buildimage/pull/5309/files#diff-775dd0b4aefdf1ad5c95b2ba8deb5b50R323